pkg/controller/sync/controller.go

```diff
@@ -452,9 +452,13 @@ func (s *FederationSyncController) objectForUpdateOp(desiredObj, clusterObj *uns
 	// Pass the same ResourceVersion as in the cluster object for update operation, otherwise operation will fail.
 	desiredObj.SetResourceVersion(clusterObj.GetResourceVersion())
 
-	if s.typeConfig.GetTarget().Kind == util.ServiceKind {
+	targetKind := s.typeConfig.GetTarget().Kind
+	if targetKind == util.ServiceKind {
```

Comment:
(no action required) maybe switch is better than multiple if's

Reply:
I'm thinking we'll want a configurable way to do this before too long, so I'm not too worried about optimizing it.